hack/verify-generated.sh

```diff
@@ -18,21 +18,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-HELM_DOCS_BIN=${1:-}
-
-if [[ -z $HELM_DOCS_BIN ]]; then
-	echo "usage: $0 <path-to-helm-docs>"
-	exit 1
-fi
-
-TEMP_FILE=$(mktemp)
-trap '{ rm -f -- "$TEMP_FILE"; }' EXIT
-
-$HELM_DOCS_BIN ./deploy/charts/trust-manager -d -l error > ${TEMP_FILE}
-
-if ! cmp -s "./deploy/charts/trust-manager/README.md" "${TEMP_FILE}"; then
-  echo "Helm chart README.md is out of date."
-  echo "Please run 'make update-helm-docs'."
+git --no-pager diff
+git status --short --untracked-files=no
+if [ ! -z "$(git status --short --untracked-files=no)" ]; then
```

Comment:
I remember that @wallrj had some issues with this git status setup.
I think at that point we settled for a script like the script here: https://github.com/cert-manager/approver-policy/blob/main/hack/util/verify.sh

And a verify target like this:
```make
verify:
	./hack/util/verify.sh $(MAKE) -s generate-all
```

Reply:
I think that the issue was as follows:
- not considering untracked-files might mean that you miss files that have to be freshly created
- considering untracked-files means that local changes/ files will cause the check to fail, which was undesirable for development.

Not sure if I remember this 100% correctly.

Reply:
I'll take a look and fix. Thanks for the pointers.

Reply:
I have now made a second attempt - based on the referenced work in approver-policy. PTAL!